inspirehep/modules/workflows/tasks/submission.py

```diff
@@ -331,8 +331,26 @@ def filter_keywords(obj, eng):
 @with_debug_logging
 def prepare_keywords(obj, eng):
     """Prepares the keywords in the correct format to be sent"""
+    def _check_keyword_should_be_left_untouched(keyword):
+        if keyword.get('schema', '') == "INSPIRE" and keyword.get('source') is None:
+            return True
+        return False
+
+    def _filter_out_classifier_keywords(keyword):
+        if keyword.get('schema', '') == "INSPIRE" and keyword.get("source", '') == 'classifier':
+            return False
+        return True
     keywords = obj.data.get('keywords', [])
     if obj.data.get('core'):
+        # if there is any INSPIRE keyword without `source: classifier` then don't do anything.
+        untouchable_keywords = [kw for kw in filter(
+            _check_keyword_should_be_left_untouched,
+            keywords
+        )]
```

Comment:
why the weird mix of a list comprehension and `filter`?
you could use simply (this will also short-circuit)
```suggestion
        untouchable_keywords = any(_check_keyword_should_be_left_untouched(kw) for kw in keywords)
```

Reply:
You are right. I'm probably to tired recently and sometimes my brain is just turning off... Sorry for that.